Material UI v5

LunaticMuch opened the pull request:
This updates material UI from v3 to v5.

Because the new material contains various breaking changes and dependencies, the PR also servers for:
- React from v16 to v17 (I avoided v18)
- `allowSyntheticDefaultImports` is set to true to be compatible with material UI
- `@types/prop-types` as required by material UI
- Couple of CSS tweaks to fit the new UI

IvanGoncharov commented:
@LunaticMuch I rebased your PR and pushed some changes on top.
Can you please review them?

LunaticMuch commented:
I made a couple of changes, padding is back as it used to be and the button is grey.
Ideally, we should make few tweaks to the entire way MUI is used, but not mandatory for sorting this PR.
I rebased from you and also changed the branch default to `main` on my fork as well. I see some jobs still failing, need to look at them.
